x-pack/plugins/security_solution/scripts/endpoint/common/connectors_services.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { KbnClient } from '@kbn/test';
+import type { AllConnectorsResponseV1 } from '@kbn/actions-plugin/common/routes/connector/response';
+import type { bodySchema } from '@kbn/actions-plugin/server/routes/create';
+import type { TypeOf } from '@kbn/config-schema';
+import type { Connector } from '@kbn/actions-plugin/server/application/connector/types';
+import { catchAxiosErrorFormatAndThrow } from './format_axios_error';
+
+/**
+ * Retrieve list of configured Connectors
+ * @param kbnClient
+ */
+export const fetchConnectorsList = async (
+  kbnClient: KbnClient
+): Promise<AllConnectorsResponseV1[]> => {
+  return kbnClient
+    .request<AllConnectorsResponseV1[]>({
+      path: '/api/actions/connectors',
+      method: 'GET',
+    })
+    .catch(catchAxiosErrorFormatAndThrow)
+    .then((response) => response.data);
+};
+
+/**
+ * Returns the first connector instance (if any) of a given type
+ * @param kbnClient
+ * @param connectorTypeId
+ */
+export const fetchConnectorByType = async (
+  kbnClient: KbnClient,
+  connectorTypeId: string
+): Promise<AllConnectorsResponseV1 | undefined> => {
+  const allConnectors = await fetchConnectorsList(kbnClient);
+
+  for (const connector of allConnectors) {
+    if (connector.connector_type_id === connectorTypeId) {
+      return connector;
+    }
+  }
+};
+
+type CreateConnectorBody = TypeOf<typeof bodySchema>;
+
+/**
+ * Creates a connector in the stack
+ * @param kbnClient
+ * @param createPayload
+ */
+export const createConnector = async (
+  kbnClient: KbnClient,
+  createPayload: CreateConnectorBody
+): Promise<Connector> => {
+  return kbnClient
+    .request<Connector>({
+      path: '/api/actions/connector',
```

Comment:
I'm guessing these path values don't have defined constants yet.`/api/actions/connectors` and `/api/actions/connector`

Reply:
I could not find one. only found one for the `BASE_ACTION_API_PATH`, so I just used the full path value instead.